test/snapshots/mac/macPackagerTest.js.snap

```diff
@@ -152,23 +168,43 @@ Object {
             "blockMapSize": "@blockMapSize",
             "sha512": "@sha512",
             "size": "@size",
-            "url": "TestApp-1.1.0-mac.zip",
+            "url": "TestApp-1.1.0-mac-x64.zip",
+          },
+          Object {
+            "blockMapSize": "@blockMapSize",
+            "sha512": "@sha512",
+            "size": "@size",
+            "url": "TestApp-1.1.0-mac-arm64.zip",
+          },
+          Object {
+            "sha512": "@sha512",
+            "size": "@size",
+            "url": "TestApp-1.1.0-mac-x64.dmg",
           },
           Object {
             "sha512": "@sha512",
             "size": "@size",
-            "url": "TestApp-1.1.0-mac.dmg",
+            "url": "TestApp-1.1.0-mac-arm64.dmg",
           },
         ],
-        "path": "TestApp-1.1.0-mac.zip",
+        "path": "TestApp-1.1.0-mac-x64.zip",
```

Comment:
So suffix for `x64` is not added, as defined in https://github.com/electron-userland/electron-builder/blob/master/packages/builder-util/src/arch.ts#L27-L29, then I guess this test is not valid? Or am I missing something?

Reply:
The other logic is for `dmg.ts` only. I'll double-check regarding this zip logic

Reply:
Okay, the .zip logic by default includes an arch name
https://github.com/electron-userland/electron-builder/blob/master/packages/app-builder-lib/src/electron/ElectronFramework.ts#L151

The reason this test is updated is due to the custom artifact name template that had to be updated in the test.
https://github.com/electron-userland/electron-builder/blob/1acc32e6c3eb566c3c4e59f9098db05cead4656e/test/src/mac/macPackagerTest.ts#L19

The reason for the test update is that a custom artifact string requires an `${arch}` value otherwise it attempts to produce 2 dmg artifacts of the same name. I recall this behavior also exists for Windows builds if one wants to bundle 32bit and 64bit separately, so I assumed this was the equivalent approach here as well.
Example code from my build scripts:
```
const buildSuffix = `-${process.platform === 'win32' ? arch : "${arch}"}${isDebug ? "-Debug" : ""}`;
const artifactTemplate = "${productName}-${buildVersion}" + buildSuffix + ".${ext}",
```

Can you please advise on what you'd expect the behavior here to be?

Reply:
Thank you for taking your time, @mmaietta. Currently,  because the old `artifactName` doesn't include `arch`, the `zip` package for macOS doesn't have arch suffix. I doesn't bother me too much if the zip package file name changes. Also, I think currently, `electron-updater` only uses DMG instead of ZIP for updating (https://github.com/electron-userland/electron-builder/issues/2199).

I just worry we might end up breaking something in the process with this change. If not, this is no big deal.

@devlar Would changing `artifactName` breaks `electron-updater`?

Reply:
@quanglam2807 I found an easy change to retain the previous naming style and only append the arch for mac arm64 builds.
https://github.com/electron-userland/electron-builder/blob/master/packages/app-builder-lib/src/platformPackager.ts#L475
Simply change that line to:
```
else if (this.platform !== Platform.MAC) {
```
It feels messy, and it's basically equivalent to the MAC platform<->arch hardcoding this PR is fixing, but I also don't want to break anything we couldn't foresee. The code comments there already exemplify previous backward-compatibility woes in issue https://github.com/electron-userland/electron-builder/issues/3510. An additional win, this approach should also be forward compatible with universal binaries, which by default should not have an arch suffix.

AFAICT, this approach also mirrors the Windows Packager logic with its universal installers too.

Reply:
Awesome! Thank you so much, @mmaietta.